package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,358 @@
+            for term in data_selector:
+                if term not in self.terms:
+                    raise KeyError(f"data_selector {term} invalid. Check the "
+                                   "EDRReader's `terms` attribute.")
+                data_dict[term] = self.auxdata[term]
+            return data_dict
+        elif isinstance(data_selector, str):
+            if data_selector not in self.terms:
+                raise KeyError(f"data_selector {data_selector} invalid. Check "
+                               "the EDRReader's `terms` attribute.")
+            data_dict = {"Time": self.auxdata["Time"],
+                         data_selector: self.auxdata[data_selector]}
+            return data_dict
+        else:
+            raise ValueError(f"data_selector of type {type(data_selector)} is "
+                             "not supported. Use list or str to indicate valid"
+                             " terms. Check the EDRReader's `terms` "
+                             "attribute.")
+
+    def calc_representative(self) -> Dict[str, np.ndarray]:
+        """ Calculate representative auxiliary value(s) from the data in
+        *frame_data*.
+        Overloaded here to accommodate the different data type. Now, this works
+        for energy data dictionaries.
+
+
+        Currently implemented options for calculating representative value are:
+
+          * `closest`: default; the value(s) from the step closest to in time
+            to the trajectory timestep
+
+          * `average`: average of the value(s) from steps 'assigned' to the
+            trajectory timestep.
+
+        Additionally, if ``cutoff`` is specified, only steps within this time
+        of the trajectory timestep are considered in calculating the
+        representative.
+
+        If no auxiliary steps were assigned to the timestep, or none fall
+        within the cutoff, representative values are set to ``np.nan``.
+
+        Returns
+        -------
+        Dict[str, ndarray]
+            Dictionary with keys of auxterms and array of auxiliary value(s)
+            'representative' for the timestep.
+        """
+        if self.cutoff == -1:
+            cutoff_data = self.frame_data
+        else:
+            cutoff_data = {key: val for key, val in self.frame_data.items()
+                           if abs(key) <= self.cutoff}
+
+        if len(cutoff_data) == 0:
+            # no steps are 'assigned' to this trajectory frame, so return
+            # values of ``np.nan``
+            value = self.auxstep._empty_data()
+        elif self.represent_ts_as == 'closest':
+            min_diff = min([abs(i) for i in cutoff_data])
+            # we don't know the original sign, and might have two
+            # equally-spaced steps; check the earlier time first
+            try:
+                value = cutoff_data[-min_diff]
+            except KeyError:
+                value = cutoff_data[min_diff]
+        elif self.represent_ts_as == 'average':
```

Comment:
I'm going to split my review here, so I'll leave this question, mostly for myself, with the hope that I'll find out the answer when I read the next half  :) 

What is the scope for moving this back into base?

Reply:
Moving it back into base should not be too difficult, `calc_representative` could be changed to first check the AuxReader format and then call different methods depending on the data format. But that would somewhat go against what a base class should do I feel. I'll think about it some more, but I might be in favour of going the other way and requiring that `calc_representative` be defined on the level of individual readers instead.